graviti/utility/lazy2.py

```diff
@@ -96,6 +144,41 @@ def _getitem(self, index: int) -> Any:
         i, j = self._offsets.get_coordinate(index)
         return self._pages[i][j]
 
+    def _setitem(self, start: int, stop: int, array: pa.Array) -> None:
+        start_i, start_j = self._offsets.get_coordinate(start)
+        stop_i, stop_j = self._offsets.get_coordinate(stop - 1)
+
+        pages = []
+        lengths = []
+
+        left_page = self._pages[start_i][:start_j]
+        left_length = len(left_page)
+        if left_length:
+            pages.append(left_page)
+            lengths.append(left_length)
+
+        pages.append(array)
+        lengths.append(len(array))
+
+        right_page = self._pages[stop_i][stop_j + 1 :]
+        right_length = len(right_page)
+        if right_length:
+            pages.append(right_page)
+            lengths.append(right_length)
+
+        self._pages[start_i : stop_i + 1] = pages
+        self._offsets.setitem(start_i, stop_i, lengths)
+
+        self._update_parent_pos(start_i, stop_i - start_i + len(pages) - 1)
```

Comment:
Why start from `start_i`? Should it start from `start_i + len(pages)`?
And why the pos_offset is calculated like this?

Reply:
Here is a bug, it should not start from `start_i`, it should start from `start_i + 1`, the left_page also needs to be updated, so it should not start from `start_i + len(pages)`

Reply:
```python
L1 = [...]
L2 = [...]
L1[start:stop] = L2
```
The items in L1[start + 1:] will shift `stop - start + len(L2) - 1`